instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java

```diff
@@ -104,6 +108,17 @@ private void mapLoggingEvent(LogRecordBuilder builder, ILoggingEvent loggingEven
       attributes.put(SemanticAttributes.THREAD_ID, currentThread.getId());
     }
 
+    if (captureCodeAttributes) {
+      StackTraceElement[] callerData = loggingEvent.getCallerData();
+      if (callerData != null && callerData.length > 0) {
+        StackTraceElement firstStackElement = callerData[0];
+        attributes.put(SemanticAttributes.CODE_FILEPATH, firstStackElement.getFileName());
+        attributes.put(SemanticAttributes.CODE_NAMESPACE, firstStackElement.getClassName());
+        attributes.put(SemanticAttributes.CODE_FUNCTION, firstStackElement.getMethodName());
+        attributes.put(SemanticAttributes.CODE_LINENO, firstStackElement.getLineNumber());
```

Comment:
`StackTraceElement#getLineNumber()` might return -1 -- in which case we probably don't want to add the attribute at all, since it represents an absent value.

Reply:
Done